File: androidApp/src/main/kotlin/br/com/qualorock/androidApp/ui/components/CityFilterBar.kt
package br.com.qualorock.androidApp.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.selection.selectable
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.annotation.StringRes
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import br.com.qualorock.androidApp.R
import design.QualORockThemeTokens
import domain.enum.City

/** design-system.md §4.2's per-hub badge treatment: active/inactive colors + the pt-BR hub label. */
data class CityFilterStyle(
    val activeColor: Color,
    val activeTextColor: Color,
    @param:StringRes val labelRes: Int,
)

private const val InactiveTintAlpha = 0.15f

/** A2 — the 4-hub color table from design-system.md §4.2, keyed by [City]. */
object CityFilterColors {
    fun styleFor(city: City): CityFilterStyle = when (city) {
        City.Vitoria -> CityFilterStyle(Color(QualORockThemeTokens.AccentPink), Color(QualORockThemeTokens.ColorBgDeep), R.string.city_vitoria)
        City.VilaVelha -> CityFilterStyle(Color(QualORockThemeTokens.AccentBlue), Color(QualORockThemeTokens.ColorBgDeep), R.string.city_vila_velha)
        City.Serra -> CityFilterStyle(Color(QualORockThemeTokens.AccentOrange), Color(QualORockThemeTokens.ColorBgDeep), R.string.city_serra)
        City.Cariacica -> CityFilterStyle(Color(QualORockThemeTokens.AccentPurple), Color.White, R.string.city_cariacica)
    }
}

/** A2 — Horizontal City Filter Bar (design-system.md §4.2). */
@Composable
fun CityFilterBar(selected: City?, onSelect: (City) -> Unit, modifier: Modifier = Modifier) {
    Row(
        horizontalArrangement = Arrangement.spacedBy(QualORockThemeTokens.Space2Dp.dp),
        modifier = modifier
            .horizontalScroll(rememberScrollState())
            .padding(horizontal = QualORockThemeTokens.Space3Dp.dp, vertical = QualORockThemeTokens.Space2Dp.dp),
    ) {
        City.entries.forEach { city ->
            CityFilterPill(city = city, isActive = city == selected, onClick = { onSelect(city) })
        }
    }
}

@Composable
private fun CityFilterPill(city: City, isActive: Boolean, onClick: () -> Unit) {
    val style = CityFilterColors.styleFor(city)
    val backgroundColor = if (isActive) style.activeColor else style.activeColor.copy(alpha = InactiveTintAlpha)
    val textColor = if (isActive) style.activeTextColor else style.activeColor

    Text(
        text = stringResource(style.labelRes).uppercase(),
        color = textColor,
        fontWeight = FontWeight.SemiBold,
        fontSize = QualORockThemeTokens.TextBadge.SizeSp.sp,
        modifier = Modifier
            .clip(RoundedCornerShape(QualORockThemeTokens.RadiusPillDp.dp))
            .background(backgroundColor)
            .selectable(selected = isActive, onClick = onClick, role = Role.RadioButton)
            .padding(horizontal = QualORockThemeTokens.Space3Dp.dp, vertical = QualORockThemeTokens.Space2Dp.dp),
    )
}

File: iosApp/iosApp/UI/Components/CityFilterBar.swift
import SwiftUI
import shared

/// design-system.md §4.2's per-hub badge treatment: active/inactive colors + the pt-BR hub label.
struct CityFilterStyle {
    let activeColor: Color
    let activeTextColor: Color
    let labelKey: String.LocalizationValue
}

private let inactiveTintAlpha = 0.15

/// I2 — the 4-hub color table from design-system.md §4.2, keyed by [City] (mirrors Android's
/// `CityFilterColors`).
enum CityFilterColors {
    static func style(for city: City) -> CityFilterStyle {
        switch city {
        case .vitoria:
            return CityFilterStyle(
                activeColor: QorColor.accentPink, activeTextColor: QorColor.bgDeep, labelKey: "city_vitoria"
            )
        case .vilavelha:
            return CityFilterStyle(
                activeColor: QorColor.accentBlue, activeTextColor: QorColor.bgDeep, labelKey: "city_vila_velha"
            )
        case .serra:
            return CityFilterStyle(
                activeColor: QorColor.accentOrange, activeTextColor: QorColor.bgDeep, labelKey: "city_serra"
            )
        case .cariacica:
            return CityFilterStyle(
                activeColor: QorColor.accentPurple, activeTextColor: .white, labelKey: "city_cariacica"
            )
        default:
            // Kotlin/Native's Objective-C export represents `City` as a class hierarchy, not a
            // true Swift enum, so `switch` requires this clause even though every real case is
            // covered above — fail loudly (mirrors the fail-loud, no-catch-all rule
            // `shared/domain/enum/City.kt` documents) rather than silently defaulting to a real
            // city's styling, which would misrepresent a genuine contract break as valid data.
            fatalError("Unhandled City case: \(city)")
        }
    }
}

/// I2 — Horizontal City Filter Bar (design-system.md §4.2).
struct CityFilterBar: View {
    let selected: City?
    let onSelect: (City) -> Void

    var body: some View {
        ScrollView(.horizontal, showsIndicators: false) {
            HStack(spacing: QorSpace.space2) {
                ForEach(City.entries, id: \.self) { city in
                    CityFilterPill(city: city, isActive: city == selected) { onSelect(city) }
                }
            }
            .padding(.horizontal, QorSpace.space3)
            .padding(.vertical, QorSpace.space2)
        }
    }
}

private struct CityFilterPill: View {
    let city: City
    let isActive: Bool
    let onClick: () -> Void

    var body: some View {
        let style = CityFilterColors.style(for: city)
        let backgroundColor = isActive ? style.activeColor : style.activeColor.opacity(inactiveTintAlpha)
        let textColor = isActive ? style.activeTextColor : style.activeColor

        Button(action: onClick) {
            Text(String(localized: style.labelKey))
                .textCase(.uppercase)
                .font(.system(size: CGFloat(QualORockThemeTokens.TextBadge.shared.SizeSp), weight: .semibold))
                .foregroundStyle(textColor)
                .padding(.horizontal, QorSpace.space3)
                .padding(.vertical, QorSpace.space2)
                .background(backgroundColor)
                .clipShape(Capsule())
                .scaleEffect(isActive ? 1.05 : 1.0)
                .animation(QorMotion.easeSmooth, value: isActive)
        }
        .buttonStyle(.plain)
        .accessibilityAddTraits(isActive ? [.isSelected] : [])
    }
}
